openlibrary/plugins/openlibrary/js/realtime_account_validation.js

```diff
@@ -50,18 +50,24 @@ export function initRealTimeValidation() {
     function validateUsername() {
         const value_username = $(this).val();
         if (value_username !== '') {
-            $.ajax({
-                url: '/account/validate',
-                data: { username: value_username },
-                type: 'GET',
-                success: function(errors) {
-                    if (errors.username) {
-                        renderError('#username', '#usernameMessage', errors.username);
-                    } else {
-                        clearError('#username', '#usernameMessage');
+            if (value_username.length < 3 || value_username.length > 20) {
+                renderError('#username', '#usernameMessage', i18nStrings['input_length_err']);
+            } else if (!(/^[A-Za-z0-9-_]{3,20}$/.test(value_username))) {
```

Comment:
It would be nice to pull out these constants / magic numbers so their function/purpose is clearer, e.g.

`VALID_USERNAME_RE`, `MIN_USERNAME`, `MAX_USERNAME`

One way to achieve this could be to swap the regex logic...

```
if (!VALID_USERNAME_RE.test(value_username))  {
  if (value_username.length < MIN_USERNAME || value_username.length > MAX_USERNAME) {
        renderError('#username', '#usernameMessage', i18nStrings['input_length_err']);
    } else {
    renderError('#username', '#usernameMessage', i18nStrings['username_char_err']);
    }
} else {
  ... // success case; clear, ajax
}
```

We can possibly ignore/simplify `value_username !== ''` because that would be caught by the min/max length, unless we specifically wanted a different error shown for this case. 🤷

Reply:
I like this method a lot! Will switch things around -- and the `value_username !== ''` right at the top is there I believe so that if someone clears their answer any error messages will be removed. 

But I'll investigate moving it around to make things a little more efficient and less nested.